lib/phoenix/live_dashboard/router.ex

```diff
@@ -98,20 +109,49 @@ defmodule Phoenix.LiveDashboard.Router do
                 ":env_keys must be a list of strings, got: #{inspect(other)}"
       end
 
+    historical_data =
+      case options[:historical_data] do
+        nil ->
+          nil
+
+        tuple when is_tuple(tuple) ->
+          case tuple do
+            {module, function, args}
+            when is_atom(module) and is_atom(function) and is_list(args) ->
+              {module, function, args}
+
+            non_mfa ->
+              raise ArgumentError, historical_data_error(non_mfa)
+          end
+
+        other ->
+          raise ArgumentError, historical_data_error(other)
```

Comment:
```suggestion
          raise ArgumentError,
                ":historical_data must be a tuple of {module, function, args}, got: #{inspect(other)}"
```

If you merge the above, you can do this change and remove the private function below.